river/View.zig

```diff
@@ -502,18 +503,30 @@ pub fn map(view: *Self) !void {
         view.pending.ssd = ssd;
     }
 
-    if (server.input_manager.defaultSeat().focused_output) |output| {
+    var output: ?*Output = null;
+    if (server.config.output_rules.match(view)) |output_name| {
+        // Check if an output matches by name
```

Comment:
Would be neat if this checked against output name and connector name.

Reply:
Sorry, if I sound stupid, but what's a connector?

Reply:
Outputs can be addressed either by connector (for example `HDMI-A-1`) or by description (like "LG Electronics LG ULTRAWIDE 0x00001F2D").

Check against both `wlr_output.name` and `wlr_output.description`. Note that the latter doesn't always exist, so check for `null` first.

Reply:
> Outputs can be addressed either by connector (for example `HDMI-A-1`) or by description (like "LG Electronics LG ULTRAWIDE 0x00001F2D").
> 
> Check against both `wlr_output.name` and `wlr_output.description`. Note that the latter doesn't always exist, so check for `null` first.

I've checked the `wlr_output.description` for my monitors, and they both include the connector; for example: `Philips Consumer Electronics Company Philips 220E AU51014013941 (DP-1)`, so you might as well just use the connector name directly.
Furthermore, `command/output.zig`'s `getOutput` function (used by command such as `riverctl focus-output`) only checks for `wlr_output.name`, so using the monitor's name doesn't work.

I've also checked `wlr/types/wlr_output.h` for alternatives, and I've found `struct wlr_output`'s `make`, `model`, and `serial` fields, but for some reason using these if they're not null in Zig crashes the compositor

Reply:
> I've checked the wlr_output.description for my monitors, and they both include the connector

That's unfortunate, however comparing against output description is still a valid use case, not everybody has a stationary setup. I think it makes sense to concatenate make, model and serial and compare against that.

Reply:
> That's unfortunate, however comparing against output description is still a valid use case, not everybody has a stationary setup. I think it makes sense to concatenate make, model and serial and compare against that.

I have no idea why, but reading these fields crashes river, at least with [this code](https://github.com/riverwm/river/files/12463232/list-outputs-cmd.patch.gz) (patch had to be gzipped because GitHub is weird)

Reply:
Have you tried not dereferencing the pointers?